plugins/modules/yum_versionlock.py

```diff
@@ -111,18 +118,26 @@ def get_versionlock_packages(self):
     def ensure_state(self, packages, command):
         """ Ensure packages state """
         rc, out, err = self.module.run_command([self.yum_bin, "-q", "versionlock", command] + packages)
+        # If no package can be found this will be written on stdout with rc 0
+        if 'No package found for' in out:
+            self.module.fail_json(msg=out)
         if rc == 0:
             return True
         self.module.fail_json(msg="Error: " + to_native(err) + to_native(out))
 
 
 def match(entry, name):
+    match = False
     m = NEVRA_RE_YUM.match(entry)
     if not m:
         m = NEVRA_RE_DNF.match(entry)
     if not m:
         return False
-    return fnmatch(m.group("name"), name)
+    if fnmatch(m.group("name"), name):
+        match = True
+    if entry.rstrip('.*') == name:
```

Comment:
I'm not sure whether this does what you expect. It removes all trailing characters from `entry` that are `.` or `*`. So `foo****...***.*` will be converted to `foo`. But `foo.123` will not be changed.

Reply:
So, this is just rstrip, not regex.
In case we have (return output from yum_versionlock list):
`httpd-devel-0:2.4.57-2.el9.*`
it will be
`httpd-devel-0:2.4.57-2.el9`. Now, we can compare it with the name passed to Ansible.

Reply:
@felixfontein ping